src/main/java/net/earthcomputer/clientcommands/command/CTickCommand.java

```diff
@@ -0,0 +1,210 @@
+        if (currentMeasurer != null) {
+            currentMeasurer._break();
+            TaskManager.removeTask(TASK_NAME);
+            currentMeasurer = null;
+        }
+    }
+
+    // see https://github.com/Earthcomputer/clientcommands/blob/c1e37665739f0e1d6aeb826b9d4e45b7adb5d876/src/main/java/net/earthcomputer/clientcommands/command/CommandTick.java#L175-L255
+    private static class TickMeasuringTask extends SimpleTask {
+
+        private static final int PERIOD = 100;
+
+        private static final Minecraft minecraft = Minecraft.getInstance();
+
+        private int tickCount = 0;
+        private long totalTickTime = 0;
+        private long startTickTime;
+        private boolean hadFirstTick = false;
+        private long firstTickStart;
+        private long lastTickStart;
+
+        private final boolean tps;
+        private final boolean forceInaccurate;
+
+        public TickMeasuringTask(boolean tps, boolean forceInaccurate) {
+            this.tps = tps;
+            this.forceInaccurate = forceInaccurate;
+        }
+
+        public void incrTickCount(int count) {
+            if (!hadFirstTick) {
+                firstTickStart = System.nanoTime();
+                hadFirstTick = true;
+            } else {
+                tickCount += count;
+            }
+        }
+
+        public void startTick() {
+            startTickTime = System.nanoTime();
+            if (!hadFirstTick) {
+                firstTickStart = startTickTime;
+                hadFirstTick = true;
+            }
+        }
+
+        public void endTick() {
+            if (hadFirstTick) {
+                totalTickTime += System.nanoTime() - startTickTime;
+                tickCount++;
+            }
+        }
+
+        @Override
+        protected void onTick() {
+            if (tickCount >= PERIOD) {
+                lastTickStart = System.nanoTime();
+                _break();
+            }
+        }
+
+        @Override
+        public void initialize() {
+            minecraft.player.displayClientMessage(Component.translatable("commands.ctick.measuring"), false);
+        }
+
+        @Override
+        public void onCompleted() {
+            if (tps) {
+                long totalTime = lastTickStart - firstTickStart;
+                double tps = 1_000_000_000D * tickCount / totalTime;
+                minecraft.player.displayClientMessage(Component.translatable("commands.ctick.tps", totalTime == 0 ? Component.translatable("commands.ctick.tps.immeasurable") : DEC_FMT.format(tps)), false);
+            } else if (forceInaccurate) {
+                long totalTime = lastTickStart - firstTickStart;
+                double mspt = totalTime / (1_000_000D * tickCount);
+                minecraft.player.displayClientMessage(Component.translatable("commands.ctick.mspt", DEC_FMT.format(mspt)), false);
+                minecraft.player.displayClientMessage(Component.translatable("commands.ctick.mspt.inaccurate"), false);
```

Comment:
This message shouldn't display if the server is clearly lagging (e.g. the observed TPS is say 10% lower than expected). It also shouldn't display if the game is sprinting.

Reply:
This was copied over from your original code which I wanted to stay as close to as possible.

Reply:
The concept of help messages didn't exist back then, and besides we can improve on code I wrote 7 years ago :)